Capped Mixing Length Turbulence

freiler opened the pull request:
<!--
Ref. #18663 
-->

freiler committed:
Add Turbulence Kernels

freiler committed:
Modifications

freiler committed:
Updates

Escudier capped mixing length

Tuning mixinglength

freiler committed:
Add capped mixing length

freiler committed:
Documentation added
Modified velocity scaling for the turbulent viscosity (Smagorinsky)
Added capping to the mixing length formulation
Added total viscosity material
Added AuxKernel for the mixing length eddy viscosity
Ref. #18663

freiler committed:
Changed Turbulent viscosity material name Ref. #18663

Fixed Format #18666

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/801394/) on 36bffbc wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/18666/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/18666/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format dfc20eba34b361f846606a16259f54919783f86e`

freiler committed:
Added tests Ref. #18666

Moved tests to modules/navier_stokes
Changed INSADMixingLengthTurbulentViscosityMaterial.md to MixingLengthTurbulentViscosityMaterial.md
Ref. #18666

Addressed GiudGiud suggested changes Ref. #18666

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/801631/) on ade2b50 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/18666/site)

This comment will be updated on new commits.

GiudGiud commented:
Thanks for the contribution, I ll review this soon!
Looked at the tests:
- rename INASDMixingLengthTurbulentViscosityMaterial.md to MixingLengthTurbulentViscosityMaterial.md
- new tests need to be moved from moose/test to moose/modules/navier_stokes/test

GiudGiud commented on modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C:
```diff
@@ -19,13 +19,17 @@ WallDistanceMixingLengthAux::validParams()
                              "proportional to the distance from the nearest wall.");
   params.addParam<std::vector<BoundaryName>>("walls", "Boundaries that correspond to solid walls");
   params.addParam<Real>("von_karman_const", 0.4, "");
+  params.addParam<Real>("von_karman_const_0", 0.09, "");
+  params.addParam<Real>("delta", 1e9, "");
```
all 3 need a description

GiudGiud commented on modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C:
```diff
@@ -62,6 +66,13 @@ WallDistanceMixingLengthAux::computeValue()
-  // Return the mixing length
-  return _von_karman_const * std::sqrt(min_sq_dist);
+
+  if (std::sqrt(min_sq_dist) / _delta <= _von_karman_const_0 / _von_karman_const)
+  {
+    return _von_karman_const * std::sqrt(min_sq_dist);
+  }
+  else
+  {
+    return _von_karman_const_0 * _delta;
+  }
```
```suggestion
  if (std::sqrt(min_sq_dist) / _delta <= _von_karman_const_0 / _von_karman_const)
    return _von_karman_const * std::sqrt(min_sq_dist);
  else
    return _von_karman_const_0 * _delta;
```

GiudGiud commented on modules/navier_stokes/src/fvkernels/INSFVMixingLengthReynoldsStress.C:
```diff
@@ -100,7 +102,8 @@ INSFVMixingLengthReynoldsStress::computeQpResidual()
 
   // Compute the eddy diffusivitiy
-  ADReal eddy_diff = velocity_gradient * mixing_len * mixing_len;
+  // ADReal eddy_diff = velocity_gradient * mixing_len * mixing_len;
```
```suggestion
```

GiudGiud commented on modules/navier_stokes/src/materials/MixingLengthTurbulentViscosityMaterial.C:
```diff
@@ -0,0 +1,70 @@
+MixingLengthTurbulentViscosityMaterial::validParams()
+{
+  InputParameters params = ADMaterial::validParams();
+  params.addClassDescription("Computes the material corresponding to the turbulent viscosity"
+                             "for the mixing length model.");
```
change description
this computes the total viscosity not just the turbulent viscosity

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/ins/wall_distance_capped_mixing_length_aux/tests:
```diff
@@ -0,0 +1,16 @@
+    input = capped_mixing_length.i
+    exodiff = capped_mixing_length_out.e
+    method = "!dbg"
+    requirement = 'The system shall be able to impose turbulent viscosity based on the capped mixing length model.'
```
phrasing is weird with impose

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests:
```diff
@@ -0,0 +1,16 @@
+    input = mixing_length_total_viscosity.i
+    exodiff = mixing_length_total_viscosity_out.e
+    method = "!dbg"
+    requirement = 'The system shall be able to calculate the total turbulent viscosity based on the capped mixing length model.'
```
say it s a material property

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_eddy_viscosity_aux/tests:
```diff
@@ -0,0 +1,16 @@
+    input = mixing_length_eddy_viscosity.i
+    exodiff = mixing_length_eddy_viscosity_out.e
+    method = "!dbg"
+    requirement = 'The system shall be able to compute the turbulent viscosity based on the capped mixing length model.'
```
say it s a variable

GiudGiud reviewed (changes requested):
ins a FE folder. Use finite_volume/ins for these tests.

freiler committed:
Updates to class descriptions Ref. #18666

Fix error in class description #18666

Format changes #18666

GiudGiud commented:
Any idea as to why your tests are diffing? Are they running fine on your machine?

freiler committed:
Fix tests error #18666

Old Test Fixes #18666

GiudGiud commented on modules/navier_stokes/include/auxkernels/INSFVMixingLengthTurbulentViscosityAux.h:
```diff
@@ -0,0 +1,39 @@
+
+#include "AuxKernel.h"
+
+// Computes the value of the eddy viscosity for the mixing length model.
+class INSFVVelocityVariable;
+
+class INSFVMixingLengthTurbulentViscosityAux : public AuxKernel
```
this is the format for whole-class header dosctring
```suggestion
class INSFVVelocityVariable;
 
 /*
   * Computes the value of the eddy viscosity for the mixing length model.
   */ 
class INSFVMixingLengthTurbulentViscosityAux : public AuxKernel
```

GiudGiud commented on modules/navier_stokes/include/auxkernels/WallDistanceMixingLengthAux.h:
```diff
@@ -10,7 +10,7 @@
 
 #include "AuxKernel.h"
-
+// Computes the mixing length for the mixing length turbulence model.
```
same

GiudGiud commented on modules/navier_stokes/include/materials/MixingLengthTurbulentViscosityMaterial.h:
```diff
@@ -0,0 +1,40 @@
+
+#include "ADMaterial.h"
+// Class to compute the total viscosity which considers molecular and
+// mixing length model turbulent viscosity.
```
same

freiler committed:
Doc Fixes #18666

Format changes #18666

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/813031/) on 77407f0 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/18666/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/18666/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 091b3c0920114d667083ef2e66ac367d022b3487`

GiudGiud commented:
@lindsayad this adds some tests that will need some functor material conversion

lindsayad commented:
:frowning_face:
